File: POOLoterias/Java/Jogo.java
package pooloterias;

import java.util.List;
import java.util.ArrayList;
import java.util.Collections;

public class Jogo implements Loteria {
    private int quantNum;
    private int quantResul;
    private int miniCart;
    private int maxiCart;
    private List<Integer> faixas;
    private List<Integer> aposta;
    private boolean invalida;
    
    @Override
    public void fazerAposta(List<Integer> aposta) {
        this.aposta = aposta;
        
        if(this.aposta.get(this.aposta.size() - 1) <= this.maxiCart && this.quantNum == this.aposta.size()) {
            System.out.print("Os números apostados são: ");
            
            for(int i = 0; i < this.aposta.size() - 1; i++) {
                System.out.print(this.aposta.get(i) + " ");
            }
            
            System.out.println(this.aposta.get(this.aposta.size() - 1) + ".");
        }
        else {
            System.out.println("Aposta Inválida!");
            this.invalida = true;
        }
    }
    
    @Override
    public void sorteio() {
        if(!this.invalida) {
            int num;
            List<Integer> sorte = new ArrayList<>();
            int cont = 0;
            
            while(cont < this.quantResul) {
                num = (int)Math.round(Math.random() * (this.maxiCart - this.miniCart) + this.miniCart);
                
                if(!sorte.contains(num)) {
                    sorte.add(num);
                    cont++;
                }
            }
            
            Collections.sort(sorte);
            
            System.out.print("Números Sorteados: ");
            
            for(int i = 0; i < this.quantResul - 1; i++) {
                System.out.print(sorte.get(i) + " ");
            }
            
            System.out.println(sorte.get(this.quantResul - 1) + ".");
            
            int acertos = 0;
            List<Integer> numCert = new ArrayList<>();
            cont = 0;
            
            while(cont < this.quantResul) {
                if(this.aposta.contains(sorte.get(cont))) {
                    numCert.add(sorte.get(cont));
                    acertos++;
                }
                cont++;
            }
            
            if(!numCert.isEmpty()) {
                System.out.print("Você acertou " + acertos + " números que foram: ");
                
                for(int i = 0; i < numCert.size() - 1; i++) {
                    System.out.print(numCert.get(i) + " ");
                }
                
                System.out.print(numCert.get(numCert.size() - 1));
                
                if(this.faixas.contains(acertos)) {
                    System.out.println(" e acertou a faixa de " + acertos + " pontos!");
                }
                else {
                    System.out.println("!");
                }
            }
            else {
                System.out.println("Você não acertou nenhum número!");
            }
        }
        else {
            System.out.println("Sorteio inválido!");
        }
    }
    
    // Construtor
    public Jogo(int quantNum, int quantResul, int miniCart, int maxiCart, List<Integer> faixas) {
        this.quantNum = quantNum;
        this.quantResul = quantResul;
        this.miniCart = miniCart;
        this.maxiCart = maxiCart;
        this.faixas = faixas;
        this.invalida = false;
    }
    
    // Getters e Setters
    public int getQuantNum() {
        return this.quantNum;
    }
    
    public int getQuantResul() {
        return this.quantResul;
    }
    
    public int getMiniCart() {
        return this.miniCart;
    }
    
    public int getMaxiCart() {
        return this.maxiCart;
    }
    
    public List<Integer> getFaixas() {
        return this.faixas;
    }
    
    public List<Integer> getAposta() {
        return this.aposta;
    }
    
    public boolean getInvalida() {
        return this.invalida;
    }
    
    public void setQuantNum(int quantNum) {
        this.quantNum = quantNum;
    }
    
    public void setQuantResul(int quantResul) {
        this.quantResul = quantResul;
    }
    
    public void setMiniCart(int miniCart) {
        this.miniCart = miniCart;
    }
    
    public void setMaxiCart(int maxiCart) {
        this.maxiCart = maxiCart;
    }
    
    public void setFaixas(List<Integer> faixas) {
        this.faixas = faixas;
    }
    
    public void setAposta(List<Integer> aposta) {
        this.aposta = aposta;
    }
    
    public void setInvalida(boolean invalida) {
        this.invalida = invalida;
    }
}

File: POOLoterias/Python/Jogo.py
from Loteria import Loteria
from random import randint

class Jogo(Loteria):
    __quantNum = 0
    __quantResul = 0
    __miniCart = 0
    __maxiCart = 0
    __faixas = tuple()
    __aposta = tuple()
    __invalida = False

    def fazerAposta(self, aposta):
        self.__aposta = aposta

        if self.__aposta[len(self.__aposta) - 1] <= self.__maxiCart and self.__quantNum == len(self.__aposta):
            print("Os números apostados são: ", end = "")

            for i in range(0, len(self.__aposta) - 1, 1):
                print("{} ".format(self.__aposta[i]), end = "")

            print("{}.".format(self.__aposta[len(self.__aposta) - 1]))
        else:
            print("Aposta inválida!")
            self.__invalida = True

    def sorteio(self):
        if not self.__invalida:
            num = 0
            sorte = list()
            cont = 0

            while cont < self.__quantResul:
                num = randint(self.__miniCart, self.__maxiCart)

                if not num in sorte:
                    sorte.append(num)
                    cont += 1

            sorte.sort()

            print("Números sorteados: ", end = "")

            for i in range(0, self.__quantResul - 1, 1):
                print("{} ".format(sorte[i]), end = "")


            print("{}.".format(sorte[self.__quantResul - 1]))

            acertos = 0
            numCert = list()
            cont = 0

            while cont < self.__quantResul:
                if sorte[cont] in self.__aposta:
                    numCert.append(sorte[cont])
                    acertos += 1
                cont += 1

            if len(numCert) > 0:
                print("Você acertou {} números que foram: ".format(acertos), end = "")

                for i in range(0, len(numCert) - 1, 1):
                    print("{} ".format(numCert[i]), end = "")

                print(numCert[len(numCert) - 1], end = "")

                if acertos in self.__faixas:
                    print(" e acertou a faixa de {} pontos!".format(acertos))
                else:
                    print("!")
            else:
                print("Você não acertou nenhum número!")
        else:
            print("Sorteio inválido!")

    def __init__(self, quantNum, quantResul, miniCart, maxiCart, faixas):
        self.__quantNum = quantNum
        self.__quantResul = quantResul
        self.__miniCart = miniCart
        self.__maxiCart = maxiCart
        self.__faixas = faixas
        self.__invalida = False

    def __del__(self):
        print("Objeto Jogo Destruído com Sucesso!")

    def getQuantNum(self):
        return self.__quantNum

    def getQuantResul(self):
        return self.__quantResul

    def getMiniCart(self):
        return self.__miniCart

    def getMaxiCart(self):
        return self.__maxiCart

    def getFaixas(self):
        return self.__faixas

    def getAposta(self):
        return self.__aposta

    def getInvalida(self):
        return self.__invalida

    def setQuantNum(self, quantNum):
        self.__quantNum = quantNum

    def setQuantResul(self, quantResul):
        self.__quantResul = quantResul

    def setMiniCart(self, miniCart):
        self.__miniCart = miniCart

    def setMaxiCart(self, maxiCart):
        self.__maxiCart = maxiCart

    def setFaixas(self, faixas):
        self.__faixas = faixas

    def setAposta(self, aposta):
        self.__aposta = aposta

    def setInvalida(self, invalida):
        self.__invalida = invalida